### PR TITLE
[cpp-pistache] add the option to fetch dependencies 

### PR DESCRIPTION
**Pull request by etherealjoy**
### PR checklist

- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [X] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `3.1.x`, `4.0.x`. Default: `master`.
- [X] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR
Add External Libraries as downloaded projects to compile the code
Add generation of petstore without recompiling the jar

@stkrwork @MartinDelille 

```java
java -jar ./openapi-generator-cli.jar config-help -g cpp-pistache-server
```
produces
```shell
CONFIG OPTIONS
	addExternalLibs
	    Add the Possibility to fetch and compile external Libraries needed by this Framework.
(Default: false)
```

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CppPistacheServerCodegen.java [hunk 1]
```diff
@@ -43,7 +43,9 @@
 
 public class CppPistacheServerCodegen extends AbstractCppCodegen {
     protected String implFolder = "impl";
-
+    protected boolean isAddExternalLibs = true;
```
Comment (etherealjoy) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CppPistacheServerCodegen.java]: @stkrwork 
Should we make this default false.
Normally when it is true we can build it out of the box.

Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CppPistacheServerCodegen.java]: @etherealjoy I suggest either merging it to 3.2.x (breaking changes without fallback) if defaults to `true` or change the default to false if we want to merge this PR into current master.

**Comment by etherealjoy** [comment]
@wing328 
Resolved the merge conflicts.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CppPistacheServerCodegen.java [hunk 2]
```diff
@@ -410,9 +418,16 @@ public String escapeQuotationMark(String input) {
...
     }
-    
+
     @Override
     public String getTypeDeclaration(String str) {
         return toModelName(str);
     }
+
+    /**
+     * specify whether external libraries will be added during the generation
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CppPistacheServerCodegen.java]: @etherealjoy I think you will need to add a proper docstring for the parameter "value" to address the following error reported by the CI:
```

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.0.1:jar (attach-javadocs) on project openapi-generator: MavenReportException: Error while generating Javadoc: Project contains Javadoc Warnings -> [Help 1]
[ERROR] 
```

**Comment by wing328** [comment]
@etherealjoy thanks. The PR has been merged into master.

**Comment by CyrilleBenard** [comment]
@etherealjoy You wrote above that the addExternalLibs default is **false** but when have a look to the generator config-help for cpp-pistache-server I see :
```
CONFIG OPTIONS
	addExternalLibs
	    Add the Possibility to fetch and compile external Libraries needed by this Framework. (Default: true)
```

Is it only a documentation issue ?

**Comment by wing328** [comment]
It was false in 3.1.x. In 3.2.0 (current master), we've enabled it as the default (and mark that as a breaking change)

**Comment by CyrilleBenard** [comment]
Ah ok, I missed that.
Nice :+1:

**Pull request merged**